### PR TITLE
Bump dependencies to support k8s 1.18

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

- Update dependencies to k8s 1.18 version
- adjust code to the new function signature
- update `violation_exceptions.txt` (more info: https://github.com/kubernetes/kube-openapi/issues/175). Currently, just add an exception for service catalog APIs as generated `openapi` is not used

Fixes https://github.com/kubernetes-sigs/service-catalog/issues/2793

### CODE REVIEW

**Comment by mszostok** [comment]
unit test failes: https://prow.k8s.io/log?id=1259491213271109633&job=pull-service-catalog-test-unit

```
# k8s.io/apimachinery/pkg/util/errors
../../../../pkg/mod/k8s.io/apimachinery@v0.18.2/pkg/util/errors/errors.go:99:10: undefined: errors.Is
note: module requires Go 1.13
```
waitng for https://github.com/kubernetes/test-infra/pull/17567 to be merged

File: Makefile [hunk 1]
```diff
@@ -58,8 +58,8 @@ endif
 
 TYPES_FILES    = $(shell find pkg/apis -name types.go)
 GO_VERSION    ?= 1.13
-GOFLAGS		   =-mod=vendor
 
+export GOFLAGS=-mod=vendor
```
Comment (MHBauer) [inline Makefile:62]: I think these should probably add on to whatever `${GOFLAGS}` is already defined and available in the environment.

File: contrib/hack/ci/run-e2e-tests.sh [hunk 2]
```diff
@@ -85,7 +87,7 @@ test::prepare_data() {
 
 test::execute() {
     shout "- Executing e2e test..."
-    pushd ${REPO_ROOT_DIR}/test/e2e/
+    pushd "${REPO_ROOT_DIR}/test/e2e/"
```
Comment (MHBauer) [inline contrib/hack/ci/run-e2e-tests.sh:90]: nice catch

File: contrib/hack/ci/run-migration-tests.sh [hunk 3]
```diff
@@ -47,6 +47,8 @@ export TB_NAMESPACE="test-broker"
 
 DUMP_CLUSTER_INFO="${DUMP_CLUSTER_INFO:-false}"
 
+GOFLAGS=-mod=vendor
```
Comment (MHBauer) [inline contrib/hack/ci/run-migration-tests.sh:50]: why no export like the others?

**Comment by mszostok** [comment]
Thanks @MHBauer for the review, I will apply fixes.

FYI: I was also waiting until this pr: https://github.com/kubernetes/test-infra/pull/17622 will be merged. Now we will have 3 pipelines to test if the current implementation works fine with 3 latest K8s versions, currently 1.16, 1.17, and 1.18. I tested that manually but IMO having pipelines for that is the best option to do not have any regression and also prove that this PR does not break anything.

File: contrib/hack/ci/run-migration-tests.sh (hunk 3 above)
Comment (mszostok) [reply contrib/hack/ci/run-migration-tests.sh:50]: done 👍

**Pull request merged**